src/Native/System.Native/pal_process.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include <stdint.h>
+
+/**
+ * These values differ from OS to OS, so make a constant contract.
+ * These values apply for the current process only
+ */
+enum class RlimitResources : int32_t
+{
+    PAL_RLIMIT_CPU          = 0,        // CPU limit in seconds
+    PAL_RLIMIT_FSIZE        = 1,        // Largest file that can be created, in bytes
+    PAL_RLIMIT_DATA         = 2,        // Maximum size of data segment, in bytes
+    PAL_RLIMIT_STACK        = 3,        // Maximum size of stack segment, in bytes
+    PAL_RLIMIT_CORE         = 4,        // Largest core file that can be created, in bytes
+    PAL_RLIMIT_AS           = 5,        // Address space limit
+    PAL_RLIMIT_RSS          = 6,        // Largest resident set size, in bytes
+    PAL_RLIMIT_MEMLOCK      = 7,        // Locked-in-memory address space
+    PAL_RLIMIT_NPROC        = 8,        // Number of processes
+    PAL_RLIMIT_NOFILE       = 9,        // Number of open files
+};
+
+/**
+ * The current and maximum resource values for the current process.
+ * These values are depict the resource according to the above enum.
+ */
+struct RLimit
+{
+    uint64_t            CurrentLimit;
+    uint64_t            MaximumLimit;
+};
+
+
+/**
+ * Get the current limit for the specified resource of the current process.
+ * Returns 0 on success; returns -1 on failure and errno is set to the error reason.
+ */
+extern "C"
+int GetRLimit(
```

Comment:
int32_t instead of int again. (I'm not going to point out all of them, just make sure to use int32_t not int in all exported signature in header and impl file.)